docs/IntegrationGuide.md

```diff
@@ -1,95 +1,306 @@
+
+1. Second case:
+    1. User has cart with item1=10€
+    1. User is redirected to Paypal payment page
+    1. In a different tab, user changes his cart to e.g. item2=20€
+    1. User completes the payment in the first tab and is redirected back to shop
+    1. Paypal approves the payment for 10€, but the real total amount of sold (shipped) items has changed to 20€
+
+    **Possible solution:** the backend has to compare total amount of the payment and total amount of payment's cart before calling `execute/payments` endpoint.
+     In case of differences, the whole payment process must be restarted.
+
+### Bad practice
+- Never delete or un-assign created payment objects **during checkout** from the cart. If required &mdash; clean up  unused/obsolete payment objects by another asynchronous process instead.
+
+## PayPal integration service API
+
+### HTTP Responses
+All endpoints accept and return data as JSON.
+
+1. Return HTTP codes on `create/payments` endpoint URL:
+- **201**: successfully created payment in PayPal and CTP updated with `approvalUrl` as custom field
+
+1. Return HTTP codes on `execute/payments` endpoint URL:
+- **201**: successfully executed payment in PayPal, created transaction in CTP
+
+1. Return HTTP codes on `paypalplus/payments/${paypalPaymentId}` endpoint URL:
+- **200**: Paypal payment found and returned as JSON response body
+
+1. Common error codes
+- **404**: resource not found by the supplied UUID/ID
+- **400**: required request parameters are missing or wrong
+- **503**: any exception which implies that request can be safely retried with the same parameters/payload again
+- **500**: unexpected/not handled exceptions
+
+Additionally, response can contain additional response body. All fields of the response body are optional. Example:
+```json
+{
+  "approvalUrl": "https://test.de",              # applicable only in case of create payment
+  "errorCode": "",                               # only in case of error and represents a unique error code
+  "errorMessage": "Parameter 'x' is missing"     # only in case of error
+  "payment": {"id":"XXX", "intent":"sale", ...}  # only in case of getting the payment object
+}
+```
+
+## Supported payment types
+
+### Default payments
+Default payments are _Credit Card_, _Paypal_ and _Invoice_. In order to create any of default payments ensure that CTP payment object's payment method ([Payment#PaymentMethodInfo#method](http://dev.commercetools.com/http-api-projects-payments.html#paymentmethodinfo)) is set to **`default`**. See also [CtpPaymentMethods](/src/main/java/com/commercetools/payment/constants/ctp/CtpPaymentMethods.java).
+
+### Installment (Ratenzahlung) payment
+
+Installment payment type requires different workflow because of especial security requirements from Paypal Plus.
+
+  1. CTP ([PaymentMethodInfo#method](http://dev.commercetools.com/http-api-projects-payments.html#paymentmethodinfo))
```

Comment:
Add when the CTP payment should be created. Here are only requirements for the payment but I miss sth like `Backend creates CTP payment object and assigns it to the cart` step.